packages/contracts/contracts/payoutStrategy/MerklePayoutStrategy/MerklePayoutStrategyImplementation.sol

```diff
@@ -71,7 +83,48 @@ contract MerklePayoutStrategyImplementation is IPayoutStrategy, Initializable {
    *
    * @param encodedDistribution encoded distribution
    */
-  function payout(bytes[] memory encodedDistribution) external override isRoundContract payable {
-    // TODO: Add payout logic
+  function batchPayout(Distribution[] calldata _distributions) external override isRoundContract payable { 
+
+    for (uint256 i = 0; i < _distributions.length; ++i) {
+        distribute(_distributions[i]);
+    }
+
+    emit BatchPayoutTriggered(msg.sender);
+  }
+
+  function distribute(Distribution calldata _distribution) public {
+
+    uint256 _index = _distribution.index;
+    address _grantee = _distribution._grantee;
+    uint256 _amount = _distribution.amount;
+    bytes32[] calldata _merkleProof = _distribution.merkleProof;
+
+    require(!hasDistributed(_index), "MerklePayout: Funds already distributed.");
+
+    bytes32 node = keccak256(abi.encodePacked(_index, _grantee, _amount));
+    require(MerkleProof.verify(_merkleProof, merkleRoot, node), "MerklePayout: Invalid proof.");
+
+    _setDistributed(_index);
+    token.safeTransfer(_claimee, _amount);
```

Comment:
handle native token transfer

Reply:
We first must have clear where the balances live!